pkg/scheduler/objects/queue.go

```diff
@@ -177,8 +177,8 @@ func NewDynamicQueue(name string, leaf bool, parent *Queue) (*Queue, error) {
 		return nil, fmt.Errorf("dynamic queue can not be added without parent: %s", name)
 	}
 	// name might not be checked do it here
-	if !configs.QueueNameRegExp.MatchString(name) {
-		return nil, fmt.Errorf("invalid queue name '%s', a name must only have alphanumeric characters, - or _, and be no longer than 64 characters", name)
+	if !configs.QueueNameRegExp.MatchString(name) || name == common.RecoveryQueue {
+		return nil, fmt.Errorf("invalid queue name '%s', a name must only have alphanumeric characters, - or _, and be no longer than 64 characters except the recovery queue root.@recovery@", name)
```

Comment:
Error message is not exact. Now we also allow `:`, `#`, `/`, and `@`.
I suggest returning different, shorter error messages. This message is too long.
1) If recovery queue: `the queue name cannot be root.@recovery@`
2) Longer than 64 chars: `queue name is longer than 64 characters`
3) If regexp fails: `invalid queue name '%s', a name must consists of alphanumeric characters and '-', '_', '#', '@', '/', ':'`

No.2) obviously affects the regexp, but I prefer simpler regexps anyway. Checking the length separately is not a big deal.